### PR TITLE
Fix ordering in star channel index/queries

### LINKED ISSUE #3500: Duplicate doc ids with * channel
Originally https://issues.couchbase.com/browse/CM-64

SG 2.1.0-14, CBS 5.0.0-2361:
Issue:
Tests using * channel are failing with duplicate doc ids
Test case:
Run the below test from the feature/sg21 branch of mobile testkit:
pytest -s --timeout 1800 --mode=cc --server-version=5.5.0-2361 --sync-gateway-version=2.1.0-14 testsuites/syncgateway/functional/tests/ --skip-provisioning -k 'test_overloaded_channel_cache[sync_gateway_channel_cache-1000-*-True-50]'

### PR DESCRIPTION
**Pull request by adamcfraser**
Fixes #3500 

Star channel queries and _all_docs queries share the same index, to minimize the number/size of indexes that need to be created for Sync Gateway.  Previously this index was incorrectly optimized for _all_docs (ordering by id), and also wasn't doing any subsequent sequence ordering for star channel queries

It's now optimized for star channel queries, since thesemay be used in performance-sensitive functionality (changes worker implementations, sg-replicate, etc), while _all_docs is intended as ad-hoc functionality only.

The existing _all_docs unit test (TestAllDocsAccessControl) wasn't being run as an integration test, because of how the code was reused for accel/non-accel testing.  Split this out to separate tests to allow running against a Couchbase bucket.

Also cleans up some remaining obsolete handling related to legacy import, including removal of the associated N1QL query.

### CODE REVIEW

**Comment by adamcfraser** [comment]
Integration tests:
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/589/
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/590/

File: rest/api_test.go [hunk 1]
```diff
@@ -1686,20 +1685,230 @@ func TestReadChangesOptionsFromJSON(t *testing.T) {
 	assert.Equals(t, options.HeartbeatMs, uint64(60000))
 }
 
-func TestAccessControl(t *testing.T) {
-	restTester := initRestTester(db.IntSequenceType, `function(doc) {channel(doc.channels);}`)
-	defer restTester.Close()
-	testAccessControl(t, restTester)
-}
+func TestAllDocsAccessControl(t *testing.T) {
```
Comment (tleyden) [inline rest/api_test.go]: While we're here, can you add a high level description of what behavior the test is validating?

File: db/query.go [hunk 2]
```diff
@@ -180,7 +168,8 @@ var QueryAllDocs = SGQuery{
 			"$sync.sequence as s, "+
 			"$sync.channels as c "+
 			"FROM `%s` "+
-			"WHERE META(`%s`).id NOT LIKE '%s' "+
+			"WHERE $sync.sequence > 0 AND "+ // Required to use the
```
Comment (tleyden) [inline db/query.go]: Looks like the comment sentence may have gotten cut off

File: rest/api_test.go (hunk 1 above)
Comment (tleyden) [reply rest/api_test.go]: Would this test fail before the fix?

File: rest/api_test.go [hunk 3]
```diff
@@ -1686,20 +1685,230 @@ func TestReadChangesOptionsFromJSON(t *testing.T) {
...
+	log.Printf("Admin response = %s", response.Body.Bytes())
+	err = json.Unmarshal(response.Body.Bytes(), &allDocsResult)
+	assert.Equals(t, err, nil)
+	assert.Equals(t, len(allDocsResult.Rows), 5)
+	assert.Equals(t, allDocsResult.Rows[0].ID, "doc1")
+	assert.Equals(t, allDocsResult.Rows[1].ID, "doc2")
 }
 
-func testAccessControl(t *testing.T, rt indexTester) {
+func TestVbSeqAllDocsAccessControl(t *testing.T) {
```
Comment (tleyden) [inline rest/api_test.go]: Ditto on high level test description

**Review by tleyden — APPROVED** [review]
LGTM - added a few comments

File: rest/api_test.go (hunk 1 above)
Comment (adamcfraser) [reply rest/api_test.go]: This test didn't fail before the fix, but validates the change made in this PR to make sequence the primary id in the AllDocs index.  

The changes to TestChangesViewBackfillStarChannel cover the original bug - with those changes, the test failed prior to the fix.

**Comment by adamcfraser** [comment]
Integration test highlighted a missing test scenario which ended up uncovering a bug.  Updated test, fixed bug.  New integration tests:
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/592/
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/593/

**Comment by adamcfraser** [comment]
The TestAllDocsOnly failure in http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/593/ looks new.  I've been able to reproduce an intermittent failure running locally - digging into it.

**Comment by adamcfraser** [comment]
Haven't been able to reliably reproduce TestAllDocsOnly failure, so attributing it to post-flush instability for the time being.  Will revisit if it's seen in functional tests, or after we have the ability to run integration tests with index drop prior to flush.

**Pull request merged**